### ISSUE TITLE
error TS2344: Type 'Modifiers' does not satisfy the constraint 'SelectModifiers'.

### ISSUE THREAD

**Issue by jdheeter** [edgedb/edgedb-js#437]
The code built by running `yarn edgeql-js` has typescript errors. 

**Schema**

Your application schema. 

```
module default {
  type Person {
    required property first_name -> str;
    property last_name -> str;
    property full_name :=
      .first_name ++ ' ' ++ .last_name
      if exists .last_name
      else .first_name;
  }

  type Movie {
    required property title -> str;
    property release_year -> int64;
    multi link actors -> Person;
    link director -> Person;
  }
};

```

**Generated EdgeQL**

not relevant, can't compile 

**Error or desired behavior**
## Error: after running tsc command
```sh
dbschema/edgeql-js/syntax/select.ts:715:51 - error TS2344: Type 'Modifiers' does not satisfy the constraint 'SelectModifiers'.

715   __cardinality__: ComputeSelectCardinality<Expr, Modifiers>;
                                                      ~~~~~~~~~

  dbschema/edgeql-js/syntax/select.ts:703:3
    703   Modifiers = Pick<Shape, SelectModifierNames>
          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    This type 
```

Expected:
the ts library generated by edgeql-js should compile as valid ts

**Versions (please complete the following information):**
 - OS: MacOS 12.5.1
 - EdgeDB version: 2.1
 - "edgedb": "^0.22.1",
 - "edgeql-js": "^1.0.0",

**Comment by jdheeter** [comment]
to repo you can just follow the steps of this tutorial https://www.edgedb.com/docs/guides/tutorials/nextjs
as soon as you generate the client library typescript will fail to compile.

**Comment by dotlouis** [comment]
Ran into the same issue.
For a temporary fix, you can change the line
```TS
 __cardinality__: ComputeSelectCardinality<Expr, Modifiers>
```
into 
```TS
 __cardinality__: ComputeSelectCardinality<Expr, SelectModifiers>
```

**Comment by jdheeter** [comment]
great thanks, is this just an issue with the latest version of the package?

**Comment by jackvane** [comment]
> Ran into the same issue. For a temporary fix, you can change the line
> 
> ```ts
>  __cardinality__: ComputeSelectCardinality<Expr, Modifiers>
> ```
> 
> into
> 
> ```ts
>  __cardinality__: ComputeSelectCardinality<Expr, SelectModifiers>
> ```

This hack breaks a number of queries I have and introduces cardinality issues.

Example 1:
<img width="616" alt="Screen Shot 2022-09-03 at 12 20 16" src="https://user-images.githubusercontent.com/101849104/188253969-eeef90df-4d67-4e68-a81e-f7ada1eb6622.png">
<img width="817" alt="Screen Shot 2022-09-03 at 12 21 46" src="https://user-images.githubusercontent.com/101849104/188253971-6809546e-90a2-4819-92a9-750c0fd87d31.png">

Example 2:
Type inference goes from returning a single user to a list of users
<img width="378" alt="Screen Shot 2022-09-03 at 12 20 38" src="https://user-images.githubusercontent.com/101849104/188253992-ec7bae35-a86d-4814-a453-c83024ae0cce.png">
<img width="464" alt="Screen Shot 2022-09-03 at 12 21 17" src="https://user-images.githubusercontent.com/101849104/188253997-736084dd-0c23-4440-bbb5-62a7568bc83c.png">

My solution was just to revert back to version 0.21.3 until this issue gets fixed. However, the issue is still there when building the NextJS bundle with ```npm run build```, so this seems like a pretty important issue to fix as I really don't want to surpress Typescript errors when building for Next.

**Comment by jakubmazanec** [comment]
I'm getting `Type 'Modifiers' does not satisfy the constraint 'SelectModifiers'.` even when I revert to `"edgedb": "0.21.3"` 🤷‍♂️

**Comment by colinhacks** [comment]
This appears to be a TS4.8 regression (unrelated to the `edgedb@0.22.1` release). Fix published in `edgedb@0.22.2`.
